### PR TITLE
Create `ProductDataContext` based on Jotai

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Use Jotai to setup a "product data context"

- Use this context to access product data on the PDP

- Use a separate Jotai store for each product to make sure that the data is
  updated correctly

## Justify why they are needed

- This implementation will make it possible to reuse it outside of the PDP

- I opted for Jotai instead of React Context because it is easier to use. I still ended up with a little more boilerplate than I would have liked; mostly to make sure we always have the latest data.

- I considered "dangerouslyForceHydrate" but decided against it since the behavior can be inconsistent with concurrent rendering

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3628** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3628?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3632** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3632?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3632?utm_source=stack-comment).

File: apps/store/src/components/ProductData/ProductDataContext.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { ReactNode, useMemo } from 'react'
+import { ProductData } from './ProductData.types'
+import { PRODUCT_DATA_ATOM } from './useProductData'
+
+type Props = {
+  children: ReactNode
+  productData: ProductData
+}
+
+export const ProductDataContext = (props: Props) => {
```
Comment (alebedev) [inline apps/store/src/components/ProductData/ProductDataContext.tsx:12]: Should we reflect in the name that it's not React.Context?  No strong opinion from my side

**Review by alebedev — APPROVED** [review]
Added via Giphy

File: apps/store/src/components/ProductData/ProductDataContext.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductData/ProductDataContext.tsx:12]: Ahh like provider maybe?

File: apps/store/src/components/ProductData/ProductDataContext.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  )
+}
+
+type HydrateProductDataProps = {
+  productData: ProductData
+  store: ReturnType<typeof createStore>
+}
+
+const HydrateProductData = (props: HydrateProductDataProps) => {
+  useHydrateAtoms([[PRODUCT_DATA_ATOM, props.productData]], { store: props.store })
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductData/ProductDataContext.tsx:29]: Just curious: why not put this at the scope of `ProductDataContext`?

**Review by guilhermespopolin — APPROVED** [review]

**Comment by robinandeer** [comment]
### Merge activity

* **Nov 23, 4:45 PM**: @robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3632).

**Pull request merged**